src/es-module-shims.js

```diff
@@ -267,15 +267,23 @@ function resolveDeps (load, seen) {
         if (cycleShell) {
           // circular shell creation
           if (!(blobUrl = depLoad.s)) {
+            function dotOrBracketAccessor(expt) {
+              const ch = depLoad.S[expt.s];
+              return (ch === '"' || ch === "'") ? `[${depLoad.S.slice(expt.s, expt.e)}]` : '.' + expt.n;
+            }
             blobUrl = depLoad.s = createBlob(`export function u$_(m){${
               depLoad.a[1].map(
-                name => name === 'default' ? `d$_=m.default` : `${name}=m.${name}`
+                expt => expt.ln ? `${expt.ln}=m${dotOrBracketAccessor(expt)}` : `d$_=m.default`
               ).join(',')
             }}${
-              depLoad.a[1].map(name =>
-                name === 'default' ? `let d$_;export{d$_ as default}` : `export let ${name}`
-              ).join(';')
-            }\n//# sourceURL=${depLoad.r}?cycle`);
+              depLoad.a[1].map(
+                expt => expt.ln ? `let ${expt.ln};` : 'let d$_;'
```

Comment:
I think we should just skip bindings entirely if there is no local name, as in we can filter them out first.

Reply:
Why don't we have to proxy-export the default (which doesn't have a local name)?